Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift

```diff
@@ -51,6 +51,35 @@ class BackendPostReceiptDataTests: BaseBackendTests {
         expect(completionCalled).toEventually(beTrue())
     }
 
+    func testPostsReceiptDataWithProductDataCorrectly() throws {
+        let path: HTTPRequest.Path = .postReceiptData
+
+        httpClient.mock(
+            requestPath: path,
+            response: .init(statusCode: .success, response: Self.validCustomerResponse)
+        )
+
+        var completionCalled = false
+
+        let isRestore = false
+        let observerMode = true
+        let productData: ProductRequestData = .createMockProductData(currencyCode: "USD")
+
+        backend.post(receiptData: Self.receiptData,
+                     appUserID: Self.userID,
+                     isRestore: isRestore,
+                     productData: productData,
+                     presentedOfferingIdentifier: nil,
+                     observerMode: observerMode,
+                     subscriberAttributes: nil,
+                     completion: { _ in
+            completionCalled = true
+        })
+
+        expect(self.httpClient.calls).toEventually(haveCount(1))
+        expect(completionCalled).toEventually(beTrue())
```

Comment:
I think this can be `expect(completionCalled) == true` since it's waiting for `expect(self.httpClient.calls).toEventually(haveCount(1))` anyway

Reply:
Yup 👍🏻

Reply:
I think this order makes more sense too:
```swift
expect(completionCalled).toEventually(beTrue())
expect(self.httpClient.calls).to(haveCount(1))
```

Reply:
Oh, yeah, good point